### PR TITLE
Add new Subscribers tab

### LINKED ISSUE #20680: Add new Subscribers tab
Parent: https://github.com/Automattic/wordpress-mobile/issues/47

### Tabs

- Traffic
- Insights
- Subscribers

### PR DESCRIPTION
**Pull request by aditi-bhatia**
Fixes https://github.com/wordpress-mobile/WordPress-Android/issues/20680

This PR adds in some basic groundwork for the new Subscribers tab, which is feature flagged behind `stats_traffic_subscribers_tab`. It is currently using `mapInsights` from `UiModelMapper.kt` as a placeholder for now, which is why you will see the screen below: 

Screenshot 2024-04-22 at 4 43 22 PM

I've left the uses cases empty for now as that work can be done in a separate PR. 

-----

## To Test:
1. With the `stats_traffic_subscribers_tab` feature flag on, ensure you see the `Subscribers` tab as it appears in the screenshot above. 
2. Accessing the `Subscribers` tab should fire the `stats_subscribers_accessed` tracked event.
3. With the `stats_traffic_subscribers_tab` feature flag off, ensure you don't see the `Subscribers` tab and the `weeks/days/months` tabs appear as expected.

-----

## Regression Notes

1. Potential unintended areas of impact

    - States related to the `stats_traffic_subscribers_tab` feature flag

4. What I did to test those areas of impact (or what existing automated tests I relied on)

    - Manual testing

5. What automated tests I added (or what prevented me from doing so)

    - Tests will be updated in future PRs as development continues

-----

## PR Submission Checklist:

- [x] I have completed the Regression Notes.
- [x] I have considered adding accessibility improvements for my changes.
- [x] I have considered if this change warrants user-facing release notes and have added them to `RELEASE-NOTES.txt` if necessary.

-----

## Testing Checklist (strike-out the not-applying and unnecessary ones):

- [ ] ~WordPress.com sites and self-hosted Jetpack sites.~
- [ ] ~Portrait and landscape orientations.~
- [ ] ~Light and dark modes.~
- [ ] ~Fonts: Larger, smaller and bold text.~
- [ ] ~High contrast.~
- [ ] ~Talkback.~
- [ ] ~Languages with large words or with letters/accents not frequently used in English.~
- [ ] ~Right-to-left languages. (Even if translation isn’t complete, formatting should still respect the right-to-left layout)~
- [ ] ~Large and small screen sizes. (Tablet and smaller phones)~
- [ ] ~Multi-tasking: Split screen and Pop-up view. (Android 10 or higher)~

### CODE REVIEW

**Comment by dangermattic** [comment]
<table>
  <thead>
    <tr>
      <th width="50"></th>
      <th width="100%" data-danger-table="true" data-kind="Warning">
          2 Warnings
      </th>
     </tr>
  </thead>
  <tbody>
    <tr>
      <td>:warning:</td>
      <td data-sticky="false"><code>strings.xml</code> files should only be updated on release branches, when the translations are downloaded by our automation.</td>
    </tr>
    <tr>
      <td>:warning:</td>
      <td data-sticky="false">PR is not assigned to a milestone.</td>
    </tr>
  </tbody>
</table>

<p align="right" data-meta="generated_by_pr-check">
  Generated by :no_entry_sign: <a href="https://danger.systems/">Danger</a>
</p>

File: WordPress/src/main/java/org/wordpress/android/ui/stats/refresh/StatsModule.kt [hunk 1]
```diff
@@ -297,6 +299,45 @@ class StatsModule {
         )
     }
 
+    /**
+     * Provides a list of use case factories that build use cases for the Traffic stats screen based on the given
+     * granularity (Day, Week, Month, Year).
+     */
```
Comment (irfano) [inline WordPress/src/main/java/org/wordpress/android/ui/stats/refresh/StatsModule.kt]: > Traffic stats screen based on the given granularity (Day, Week, Month, Year).

Copy&paste mistake. We need to update the comment according to the SUBSCRIBERS tab.

**Review by irfano — APPROVED** [review]
I corrected a copy&paste mistake (2112f8f75c13f9de90e4b54dfed22f7ffc92aec2).
LGTM! 👍🏻 Good decision to leave other stuff related to use cases to the following PRs.

**Comment by irfano** [comment]
I'm merging but I want to remind you that you can complete the "Regression Notes", "PR Submission Checklist" and "Testing Checklist" sections after the merge, @aditi-bhatia.

**Comment by aditi-bhatia** [comment]
Thank you for the reminder! I've gone ahead and updated those sections.

**Pull request merged**